dashboard/app/controllers/api/v1/schools_controller.rb

```diff
@@ -7,4 +7,14 @@ def index
     end
     render json: serialized_schools
   end
+
+  # GET /dashboardapi/v1/schoolsearch/:q/:limit
+  def search
+    limit = [40, Integer(params[:limit])].min
+    schools = School.where("name LIKE ?", "%#{params[:q]}%").order(:name).limit(limit)
```

Comment:
This isn't the final solution. This is just the plumbing and a basic implementation. An in-memory cache will replace this before we start using it in prod.